tests/helpers.py

```diff
@@ -128,10 +128,14 @@ def create_pod(
     pod_data['metadata']['namespace'] = namespace
     if pvc_name:
         pod_data['spec']['volumes'][0]['persistentVolumeClaim']['claimName'] = pvc_name
+
     if node_name:
         pod_data['spec']['nodeName'] = node_name
     else:
-        del pod_data['spec']['nodeName']
```

Comment:
you can replate try and except here with:

```python
if 'nodeName' in data['spec']:
    del pod_data['spec']['nodeName']
```

Reply:
Done

Reply:
Not sure I understand this change.
pod_data['spec']['nodeName'] is part of the yaml file by default and in case you want to bind the pod to run on specific node, you assign the node name to this attribute in the yaml file.
How this change fixes the issue of not providing node_name?

CC: @sidhant-agrawal @ebenahar

Reply:
@RazTamir 
pod_data['spec']['nodeName'] is not a part of standard pod.yaml which current mostly of the tests are using from ocs_ci/templates/CSI/rbd/pod.yaml and ocs_ci/templates/CSI/cephfs/pod.yaml

Reply:
ok thanks